File: src/callable.rs
use crate::{interpreter::{Interpreter, RuntimeError}, literal::Literal, env::Env};

pub trait Callable {
  fn call(&self, interpreter: &mut Interpreter, argv: Vec<Literal>) -> Result<Literal, RuntimeError>;
  fn arity(&self) -> Result<u8, RuntimeError>;
}

File: js/src/old/callable.ts
import type { Interpreter } from './interpreter';
import type { Literal } from './token';

export abstract class Callable {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  abstract call(interpreter: Interpreter, argv: Array<Literal>): any;
  abstract arity(): number
}
